### ISSUE TITLE
[impot sur le revenu] Calcul barème standard : ne pas utiliser le taux marginal moyen mais individualisé

### ISSUE THREAD

**Issue by johangirod** [betagouv/mon-entreprise#1885]
Aujourd'hui, le calcul du taux marginal lors de l'option "méthode barème standard" est effectué comme suis : 
> `Impôt(Somme des revenus) / Somme des revenus`

C'est une approximation qui marche si : 
- Il n'y a pas d'autres revenus dans le foyer fiscal
- Les autres revenus sont ceux d'un conjoint qui gagne à peu près la même chose

Mais dans le cas où les revenus sont différents, ou si l'activité est une activité additionnelle (exemple : auto-entrepreneur par dessus une activité de salarié), il semble plus correct d'utiliser le taux individualisé. 

Pour cela, on calcul d'abord le taux du revenu "base" (à déterminer : le plus faible si il s'agit de deux personnes d'un foyer aux revenus différent ou le revenu de l'activité salarié de base dans le cas d'une activité additionnelle) 
> `Taux base = Impôt(Revenu base) / Somme des revenus`
> `Taux additionnel = [ Impôt(Somme des revenus) - Taux base * Revenu base ] / (Somme des revenus - Revenu base)

> Détails : https://bofip.impots.gouv.fr/bofip/11256-PGP.html/identifiant%3DBOI-IR-PAS-20-20-20-20200515

**Comment by mquandalle** [comment]
Je ne comprends pas le problème avec la formule

> Impôt(Somme des revenus) / Somme des revenus

Dans la mesure où Somme des revenus inclus à la fois les revenus de la situation courante (salaire, rémunération gérant) et les « autres revenus du foyer fiscal ». Donc pour moi on ne fait pas d'approximation on calcule effectivement l'impôt avec les bons montants `Impôt(Somme des revenus)` et on retrouve ensuite le bon « taux moyen d'imposition ».

**Comment by johangirod** [comment]
Pas de problème en soi avec cette formule, mais elle calcul effectivement le taux moyen et non le taux individualisé. 

Prenons le cas d'un couple sans enfants avec une personne A avec 20 000 € de revenu imposable et une autre B avec 60 000 €.

Si B veut connaître son taux d'imposition, avec le taux moyen, cela revient à 14,8 % d'imposition. Mais, le taux est commun aux deux personnes du foyer, ce qui veut dire que la personne A va également payer à hauteur de 14,8 % d'impôt sur son salaire. Alors que si elle avait été seule, ce taux aurait été bien plus bas : seulement 4,3 %. Pour être plus juste, et appliquer la progressivité de l'impôt dans le couple, il est possible d'opter pour un taux individualisé.

Le principe est de reporter l'augmentation du taux d'imposition sur la part des revenus qui en sont "responsables". On calcule donc le taux du revenu le plus bas (ici 4,3 %). Ensuite, le taux additionnel est calculé avec la formule ci-dessus. Dans notre cas, cela donne : 
```
(80 000 € * 14,8 % - 20 000 € *  4,3 %) / (80 000 € - 20 000 €) = 19 %
```

Le taux individualisé de la personne B est ainsi bien plus proche de celui qu'elle aurait payé si elle était seule et correspond de ce fait mieux à son activité.

C'est le même précepte qui peut s'appliquer pour estimer l'imposition d'une activité complémentaire. Imaginons que je sois salarié et que je souhaite estimer le revenu supplémentaire que m'apportera une activité de loueur de meublé en autoentrepreneur. Pour ce calcul, la part d'impôt due à mes revenus salariée ne devrait pas changer avec ma nouvelle activité. Je veux ainsi reporter la totalité de l'augmentation de mon impôt due à la nouvelle activité sur le calcul du revenu net après impôts de cette activité. Grâce au taux individualisé, je peux avoir cette estimation.

Bien sûr, il devra être possible de choisir son mode de calcul préféré.

**Comment by mquandalle** [comment]
Ah d'accord. Mais on est d'accord que 

1. le recours au taux individualisé ne change pas le montant de l'impôt, juste la répartition entre les conjoints
2. le taux individualisé s'applique via une option exercée sur impots.gouv.fr dans la rubrique sur le prélèvement à la source. Par défaut c'est le même taux moyen qui s'applique sur les 2 salaires des conjoints.

Du coup l'idée serait plutôt d'ajouter une nouvelle méthode de calcul que de remplacer l’existante ?

**Comment by johangirod** [comment]
> le recours au taux individualisé ne change pas le montant de l'impôt, juste la répartition entre les conjoints

Oui, tout à fait. Par contre, il change le montant affiché dans les simulateurs.

> le taux individualisé s'applique via une option exercée sur impots.gouv.fr dans la rubrique sur le prélèvement à la source. Par défaut c'est le même taux moyen qui s'applique sur les 2 salaires des conjoints.
Du coup l'idée serait plutôt d'ajouter une nouvelle méthode de calcul que de remplacer l’existante ?

C'est bien cela. Ceci étant dit, dans le cas des simulateurs, on voudra peut-être utiliser le taux individualisé par défaut (à vérifier lors d'entretiens ?)
